agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -61,27 +67,66 @@ public static Configuration create(Path agentJarPath) throws IOException {
         Configuration config = loadConfigurationFile(agentJarPath);
         PreviewConfiguration preview = config.instrumentationSettings.preview;
 
-        preview.roleName = overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
+        preview.roleName = overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
         preview.roleInstance =
-                overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
+                overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
 
+        loadSamplingRateEnvVar(preview);
```

Comment:
i think if we change
`public FixedRateSampling fixedRate;`
to
`public FixedRateSampling fixedRate = new FixedRateSampling();`

this this can be simpler:

```suggestion
        preview.sampling.fixedRate.percentage = overlayWithEnvVar(APPLICATIONINSIGHTS_SAMPLING_RATE, preview.sampling.fixedRate.percentage);
```

Reply:
This is not clear to me.  percentage might be null? percentage is Double, this overlay method returns string.

Reply:
oh yes, maybe overload `overlayWithEnvVar`:

```
    static Double overlayWithEnvVar(String name, Double defaultValue) {
        String value = getEnv(name);
        if (value != null && !value.isEmpty()) {
            return Double.parseDouble(value);
        }
        return defaultValue;
    }
```